### ISSUE TITLE
Catching GeoTiff/WebGL errors doesn't work

### ISSUE THREAD

**Issue by m-mohr** [openlayers/openlayers#14926]
**Describe the bug**
I'm trying to catch errors when rendering a GeoTiff source via a WebGLTile layer, e.g. CORS issues.
The error is shown in the console, but it seems there's no way to catch / handle it.

**To Reproduce**
Example: <https://codesandbox.io/s/cog-forked-492cf3?file=/main.js>

**Expected behavior**
The fallback gets called due to the error.

**Comment by mike-000** [comment]
There are no errors.  EPSG:32610 is not defined so the tiles are rendered at those coordinates in the view projection (over North Africa).

**Comment by m-mohr** [comment]
Interesting, my Firefox reports a CORS error / AggregateError.
In Chrome it "works" (as in: no error, but of course there's still the projection issue).
I'd still expect that the error is handled in FF...

**Comment by mike-000** [comment]
Firefox is sensitive to server setup, see #14817
To check for errors listen for source `change` events then check `source.getState()`.

**Comment by m-mohr** [comment]
Thanks, that works. That is somewhat unintuitive though.
I'd expect that if the source.getState() returns error, that it should trigger an error event, right?
Is that something worth creating a PR for?

**Comment by mike-000** [comment]
> I'd expect that if the source.getState() returns error, that it should trigger an error event, right?

No, the TileJSON and BingMaps sources also set the state to `error` with only a `change` event if they fail to become `ready`.

**Comment by mike-000** [comment]
> Firefox is sensitive to server setup, see #14817

I have just checked the new https://openlayers.org/en/main/examples/cog-projection.html example and it was not working on Firefox.  Adding `Header set Access-Control-Allow-Headers "range"` to the server's `.htaccess` fixed it.

**Comment by m-mohr** [comment]
Thanks, but as a user, I'm completely confused. What is the error event for and when does it apply? If the change event thing is intentional, then it at least needs better documentation.

**Comment by mike-000** [comment]
I think that is a separate issue.  A large number of classes inherit the `error` event from `ol/Observable` but it is rarely used (all I can find is `ol/layer/MapboxVector`, which will be removed by #14870).  The API for `ol/Geolocation` shows two error events, its own and the inherited one.  If not used it should be removed.
